src/k8s-configuration/azext_k8s_configuration/custom.py

```diff
@@ -81,22 +77,25 @@ def create_k8s_configuration(client, resource_group_name, cluster_name, name, re
         helm_operator_properties.chart_version = helm_operator_chart_version.strip()
         helm_operator_properties.chart_values = helm_operator_params.strip()
 
-    protected_settings = validate_and_get_protected_settings(ssh_private_key,
-                                                             ssh_private_key_file,
-                                                             https_user,
-                                                             https_key)
-    knownhost_data = get_data_from_key_or_file(ssh_known_hosts, ssh_known_hosts_file)
+    protected_settings = _get_protected_settings(ssh_private_key,
+                                                 ssh_private_key_file,
+                                                 https_user,
+                                                 https_key)
+    knownhost_data = _get_data_from_key_or_file(ssh_known_hosts, ssh_known_hosts_file)
     if knownhost_data:
-        validate_known_hosts(knownhost_data)
+        _validate_known_hosts(knownhost_data)
 
     # Flag which parameters have been set and validate these settings against the set repository url
     ssh_private_key_set = ssh_private_key != '' or ssh_private_key_file != ''
     known_hosts_contents_set = knownhost_data != ''
     https_auth_set = https_user != '' and https_key != ''
-    validate_url_with_params(repository_url,
-                             ssh_private_key_set=ssh_private_key_set,
-                             known_hosts_contents_set=known_hosts_contents_set,
-                             https_auth_set=https_auth_set)
+    _validate_url_with_params(repository_url,
+                              ssh_private_key_set=ssh_private_key_set,
+                              known_hosts_contents_set=known_hosts_contents_set,
+                              https_auth_set=https_auth_set)
+
+    # Validate that the subscription is registered to Microsoft.KubernetesConfiguration
+    _validate_cc_registration(cmd)
```

Comment:
I'm curious why we need to do this validation?  Is the default error from ARM not sufficient to explain to the customer?

Reply:
There was discussion among Safeer and the IDC about adding this to give an explicit error back to the user and clear instructions about where to go if this validation fails